test/e2e/apimachinery/aggregator.go

```diff
@@ -57,38 +57,24 @@ const (
 )
 
 var _ = SIGDescribe("Aggregator", func() {
-	var ns string
-	var c clientset.Interface
 	var aggrclient *aggregatorclient.Clientset
 
-	// BeforeEachs run in LIFO order, AfterEachs run in FIFO order.
-	// We want cleanTest to happen before the namespace cleanup AfterEach
-	// inserted by NewDefaultFramework, so we put this AfterEach in front
-	// of NewDefaultFramework.
-	ginkgo.AfterEach(func() {
-		cleanTest(c, aggrclient, ns)
-	})
-
 	f := framework.NewDefaultFramework("aggregator")
 	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
 
 	// We want namespace initialization BeforeEach inserted by
 	// NewDefaultFramework to happen before this, so we put this BeforeEach
 	// after NewDefaultFramework.
 	ginkgo.BeforeEach(func() {
-		c = f.ClientSet
-		ns = f.Namespace.Name
-
-		if aggrclient == nil {
-			config, err := framework.LoadConfig()
-			if err != nil {
-				framework.Failf("could not load config: %v", err)
-			}
-			aggrclient, err = aggregatorclient.NewForConfig(config)
-			if err != nil {
-				framework.Failf("could not create aggregator client: %v", err)
-			}
```

Comment:
I thought already commented, this change is unrelated to the contexts, right? can we split it out?

Reply:
It's related because `ginkgo.DeferCleanup` can handle functions which take a context, i.e. this change became necessary when cleanTest was changed to require a contect.

It can be split out, but then it has to get merged before this PR.

Reply:
=> https://github.com/kubernetes/kubernetes/pull/114392